drainer/schema.go

```diff
@@ -484,3 +492,174 @@ func addImplicitColumn(table *model.TableInfo) {
 func skipJob(job *model.Job) bool {
 	return !job.IsSynced() && !job.IsDone()
 }
+
+// learn from handlePreviousDDLJobIfNeed
+func (s *Schema) handlePreviousSchemasIfNeed(version int64) error {
+	// start from 1 is ok?
+	v := int64(1)
+	for key, info := range s.dbInfos {
+		if v > version {
+			return nil
+		}
+		if v <= s.currentVersion {
+			log.Warn("schema version is less than current version, skip this schema",
+				zap.String("db", key.schemaName),
+				zap.String("stmt", info.stmt),
+				zap.Int64("id", info.id),
+				zap.Int64("currentVersion", s.currentVersion))
+			continue
+		}
+		if job, err := s.mockCreateSchemaJob(info.stmt, info.id, v); err != nil {
+			log.Error("fail to handle create schema", zap.Error(err))
+			return err
+		} else if _, _, _, err = s.handleDDL(job); err != nil {
+			return err
+		}
+		v++
+	}
+
+	for key, info := range s.tbInfos {
+		if v > version {
+			return nil
+		}
+		if v <= s.currentVersion {
+			log.Warn("schema version is less than current version, skip this schema",
+				zap.String("db", key.schemaName),
+				zap.String("tb", key.tableName),
+				zap.String("stmt", info.stmt),
+				zap.Int64("id", info.id),
+				zap.Int64("currentVersion", s.currentVersion))
+			continue
+		}
+		dbInfo, ok := s.dbInfos[schemaKey{schemaName: key.schemaName}]
+		if !ok {
+			return errors.Errorf("schema %s not found", key.schemaName)
+		}
+		if job, err := s.mockCreateTableJob(info.stmt, dbInfo.id, info.id, v); err != nil {
+			log.Error("fail to handle create table", zap.Error(err))
+			return err
+		} else if _, _, _, err = s.handleDDL(job); err != nil {
+			return err
+		}
+		v++
+	}
+	return nil
```

Comment:
Add this to make sure these memory can be garbage collected?
```suggestion
	// safe for gc
	s.dbInfos = nil
	s.tbInfos = nil
	return nil
```